### ISSUE TITLE
Catalina Errors

### ISSUE THREAD

**Issue by nemasu** [rmc-team/macos-downloader#5]
Hello, I get the following errors, the installer is invalid if I try to use it.

```
nemasus-iMac:release-205056-180919 nemasu$ sudo ./macOS\ Downloader.sh 
+ Checked system environment.
+ Root permissions check passed.
+ Resources check passed.
+ Integrity conectivity check passed.
/ What save folder would you like to use?
/ Input a save folder path.
/ /Users/nemasu/Downloads/
/ What operation would you like to run?
/ Input an operation number.
/     1 - Catalina
/     2 - Mojave
/     3 - High Sierra
/ 1
/ What operation would you like to run?
/ Input an operation number.
/     1 - Installer
/     2 - Update
/ 1
+ Downloaded installer files.
> Preparing installer.
chmod: ./resources/pbzx: No such file or directory
./macOS Downloader.sh: line 245: ./resources/pbzx: No such file or directory
cp: /Users/nemasu/Downloads//Install macOS Catalina Beta.app/Contents/SharedSupport/InstallESD.dmg: No such file or directory
cp: /Users/nemasu/Downloads//Install macOS Catalina Beta.app/Contents/SharedSupport: No such file or directory
cp: /tmp/RecoveryHDMeta/: unable to copy extended attributes to /Users/nemasu/Downloads//Install macOS Catalina Beta.app/Contents/SharedSupport: No such file or directory
cp: /Users/nemasu/Downloads//Install macOS Catalina Beta.app/Contents/SharedSupport/AppleDiagnostics.chunklist: No such file or directory
cp: /Users/nemasu/Downloads//Install macOS Catalina Beta.app/Contents/SharedSupport/AppleDiagnostics.dmg: No such file or directory
cp: /Users/nemasu/Downloads//Install macOS Catalina Beta.app/Contents/SharedSupport/BaseSystem.chunklist: No such file or directory
cp: /Users/nemasu/Downloads//Install macOS Catalina Beta.app/Contents/SharedSupport/BaseSystem.dmg: No such file or director+ Prepared installer.
+ Removed temporary files.
/ Thank you for using macOS Downloader.

```

**Comment by maelcum** [comment]
tl;dr
The URLs for macOS Catalina are not working. Can be verified by checking the Catalog.sh and trying to manually download any of the files given there.
e.g: http://swcdn.apple.com/content/downloads/53/58/061-96006-A_D2HTVCGUD8/gdt4thee08sjbckqx4p9efpww12qgz3w98


long read (more a log of my thoughts, no revelation here):

I have the exact same error (which is identical to issue #9) with release 2.3.7 as well as the current 2.4.1

The problem here is that the step "Downloaded installer files." has *not* finished successfully, although there is no error. The following error messages about ./resources/pbzx etc. are not relevant (except as a pointer that something is wrong).

I am not an expert, but tried to understand the script. I find it clear and concise, although I do not understand everything (e.g. why the function Check_Resources() is called twice; looks unnecessary to me).

If my observations are correct, it is the function Download_Installer() where things fail. Comparing the respective part of the script in both versions 2.3.7 and 2.4.1 shows that you've gone to great length to make sure curl will work, but the logic is the same. That explains why the error is happening in both versions.

The function Download_Installer():
- creates folder /tmp/$installer_name. << works
- writes something to /tmp/$installer_name/Catalog.sh. << works
- compares $InstallAssistantAuto_pkg to 1. $InstallAssistantAuto_pkg has not been assigned yet. so the comparison fails (<< works) and
- curl downloads InstallAssistantAuto.pkg to /tmp/$installer_name/InstallAssistantAuto.pkg. << this fails (as does every other download)

The problem is, that the output of curl (which is saying that the file can not be found) is redirected to the respective filename (like InstallAssistantAuto.pkg) and thus it looks as everything is fine, albeit the files being only 9 bytes in length -> the 9 characters of "Not found".

Downloading Mojave works, since the URLs are correct, the files can be downloaded and every other step that relies on the files works fine as well. It's the Catalina URLs that do not work. It has nothing to do with pbxz not being found or not being executable etc.

**Comment by maelcum** [comment]
Update: The correct URL for the example above would be: http://swcdn.apple.com/content/downloads/35/40/061-96006-A_3D42K49AN3/r1513y9zscckrsmy018hbcryom8nq7gg1t/InstallAssistantAuto.pkg

As per the Apple Download index: https://swscan.apple.com/content/catalogs/others/index-10.15-10.14-10.13-10.12-10.11-10.10-10.9-mountainlion-lion-snowleopard-leopard.merged-1.sucatalog

**Comment by julian-fairfax** [comment]
Fixed them. It seems like the new installer has a lower version number than the one I had before. Odd.
